src/backend/graphItemService.ts

```diff
@@ -1,153 +1,265 @@
+    csv.split(/\r?\n/).forEach((row) => {
+      const values = row.split(',').map((value) => {
+        if (value === '') {
+          return null; // Cypher ignores null properties when creating
+        }
+        return value;
+      });
+      if (headers === undefined) {
+        headers = values.slice(1);
+      } else {
+        total++;
+        const [UID, ...properties] = values;
+        if (batchedRows[0].rows.length === 0) {
+          batchedRows[0].rows.push(properties);
+          batchedRows[0].UIDs.push(UID + '');
+          batchedRows.push({rows: [], UIDs: []});
+        } else if (batchedRows[batchedRows.length - 1].rows.length === MAX_BATCH_SIZE) {
+          batchedRows.push({rows: [properties], UIDs: [UID + '']});
+        } else {
+          batchedRows[batchedRows.length - 1].rows.push(properties);
+          batchedRows[batchedRows.length - 1].UIDs.push(UID + '');
+        }
       }
-    }
-    return properties;
+    });
+    GraphItemService.checkNonEmptyHeaders(headers);
+    return {
+      total: total,
+      headers: headers,
+      batchedRows: batchedRows
+    };
   }
 
   /**
-   * Return the graph ID of a node given its uid and category
+   * TODO take care of escaping commas, line-breaks, etc...
+   * EOL is \n on POSIX and \r\n on Windows
    */
-  private static async getNodeID(
-    rc: RestClient,
-    sourceKey: string,
-    category: string,
-    uid: string | undefined
-  ): Promise<string> {
-    // TODO we can cache the nodes on creation and get them here
-    if (!uid) {
-      throw new Error(RowErrorMessage.SOURCE_TARGET_NOT_FOUND);
-    }
-    const query = `MATCH (n:\`${category}\`) WHERE n.UID = "${uid}" return n limit 1`; // TODO escape properly
-    try {
-      const res = await rc.graphQuery.runQuery({
-        sourceKey: sourceKey,
-        query: query
+  public static batchEdges(csv: string): {
+    total: number;
+    headers: string[];
+    batchedRows: {rows: unknown[][]; UIDs: string[]}[];
+  } {
+    const MAX_BATCH_SIZE = 500;
+    let total = 0;
+    let headers: unknown[] | undefined = undefined;
+    let batchedRows: {rows: unknown[][]; UIDs: string[]}[] = [{rows: [], UIDs: []}];
+    csv.split(/\r?\n/).forEach((row) => {
+      const values = row.split(',').map((value) => {
+        if (value === '') {
+          return null; // Cypher ignores null properties when creating
+        }
+        return value;
       });
-      if (res.isSuccess() && res.body.nodes.length > 0) {
-        // We return the id if the was found, in any other case we fail with SOURCE_TARGET_NOT_FOUND
-        return res.body.nodes[0].id;
+      if (headers === undefined) {
+        headers = values.slice(2);
+      } else {
+        total++;
+        let [from, to, ...properties] = values;
+        // TODO exclude from batch if its extremities are not found
+        properties = [
+          GraphItemService.nodeIDS.get(from + '')!,
```

Comment:
With this condition we force the user to always load the nodes before the edges (and not restart the plugin before that).
If we keep it we need to add it in the readme.
we can also suppose that the user will fill the first two column of the edge csv with the graph ids of the nodes in that case if `GraphItemService.nodeIDS.get(from + '') `return null we use the value in the file.

Reply:
yeah we need to add it to the readme because that's a consequence of changing the strategy from merging to creating